core/src/main/java/com/datastax/oss/driver/internal/core/protocol/ByteBufPrimitiveCodec.java

```diff
@@ -220,22 +222,6 @@ public void writeShortBytes(byte[] bytes, ByteBuf dest) {
     dest.writeBytes(bytes);
   }
 
-  // Reads *all* readable bytes from a buffer and return them.
-  // If the buffer is backed by an array, this will return the underlying array directly, without
-  // copy.
-  private static byte[] readRawBytes(ByteBuf buffer) {
-    if (buffer.hasArray() && buffer.readableBytes() == buffer.array().length) {
-      // Move the readerIndex just so we consistently consume the input
-      buffer.readerIndex(buffer.writerIndex());
-      return buffer.array();
-    }
```

Comment:
@chibenwa I'm not sure I understand.  Is there a reason why we don't want to use the underlying byte array directly if we can get access to it?

Reply:
Apologies @chibenwa, I've been out for the holiday here... just getting back to this.

Yeah, your explanation makes sense, thanks for the additional context.  I'm reluctant to give up the re-use of the backing array, though, so I put together an [alternate implementation](https://github.com/datastax/java-driver/pull/1621) which preserves that feature.  What do you think of something in that vein?

Reply:
The alternate proposal never went anywhere so we're moving forward with the original proposal from @chibenwa